src/ProxyManager/ProxyGenerator/LazyLoadingValueHolder/MethodGenerator/MagicGet.php

```diff
@@ -42,33 +42,70 @@ class MagicGet extends MagicMethodGenerator
      * @param PropertyGenerator   $initializerProperty
      * @param PropertyGenerator   $valueHolderProperty
      * @param PublicPropertiesMap $publicProperties
+     *
+     * @throws \Zend\Code\Generator\Exception\InvalidArgumentException
+     * @throws \InvalidArgumentException
      */
     public function __construct(
         ReflectionClass $originalClass,
         PropertyGenerator $initializerProperty,
         PropertyGenerator $valueHolderProperty,
         PublicPropertiesMap $publicProperties
     ) {
-        parent::__construct($originalClass, '__get', [new ParameterGenerator('name')]);
-
-        $this->setDocblock(($originalClass->hasMethod('__get') ? "{@inheritDoc}\n" : '') . '@param string $name');
+        $hasParent = $originalClass->hasMethod('__get');
 
         $initializer = $initializerProperty->getName();
         $valueHolder = $valueHolderProperty->getName();
         $callParent  = 'if (isset(self::$' . $publicProperties->getName() . "[\$name])) {\n"
             . '    return $this->' . $valueHolder . '->$name;'
             . "\n}\n\n";
 
-        $callParent .= PublicScopeSimulator::getPublicAccessSimulationCode(
-            PublicScopeSimulator::OPERATION_GET,
-            'name',
-            null,
-            $valueHolderProperty
+        if ($hasParent) {
+            $parameter = $originalClass->getMethod('__get')->getParameters()[0];
+
+            $parameterType = $parameter->hasType() ? $parameter->getType() : null;
```

Comment:
This is a fix for #346, no? Can you please revert it, and we move it to something else? We'd have to check it for all scenarios anyway...